Sources/CoffeeAddictsLibrary/Helpers/CoffeeShopsCSVDownloader.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  File.swift
+//  
+//
+//  Created by Mihai Garda Popescu on 23.07.2021.
+//
+
+import Foundation
+
+typealias CSVResult = (String?, ParsingStatus) -> Void
+
+class CoffeeShopsCSVDownloader {
+    static func downloadCSVFileFromURLString(_ urlString: String, completion: @escaping CSVResult) {
+        let sema = DispatchSemaphore(value: 0)
+
+        let documentsUrl:URL =  (FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first as URL?)!
+        let destinationFileUrl = documentsUrl.appendingPathComponent(Constants.localCSVFileName)
+        
+        do {
```

Comment:
I think this method has too much logic in it. Can we break it in smaller functions?

Reply:
Done